lib/src/commands/OptionsProcessor.ts

```diff
@@ -12,34 +12,50 @@ import { ColorService } from '../adapters/ColorService';
 import { AssetService } from '../adapters/AssetResolver';
 import { Options } from '../interfaces/Options';
 import { Deprecations } from './Deprecations';
+import { OptionProcessorsRegistry } from 'react-native-navigation/processors/OptionProcessorsRegistry';
 
 export class OptionsProcessor {
   constructor(
     private store: Store,
     private uniqueIdProvider: UniqueIdProvider,
+    private optionProcessorsRegistry: OptionProcessorsRegistry,
     private colorService: ColorService,
     private assetService: AssetService,
     private deprecations: Deprecations
   ) {}
 
-  public processOptions(options: Options) {
-    this.processObject(options, clone(options), (key, parentOptions) => {
-      this.deprecations.onProcessOptions(key, parentOptions);
-    });
+  public processOptions(options: Options, commandName: string) {
+    this.processObject(
+      options,
+      clone(options),
+      (key, parentOptions) => {
+        this.deprecations.onProcessOptions(key, parentOptions);
+      },
+      commandName
+    );
   }
 
-  public processDefaultOptions(options: Options) {
-    this.processObject(options, clone(options), (key, parentOptions) => {
-      this.deprecations.onProcessDefaultOptions(key, parentOptions);
-    });
+  public processDefaultOptions(options: Options, commandName: string) {
+    this.processObject(
+      options,
+      clone(options),
+      (key, parentOptions) => {
+        this.deprecations.onProcessDefaultOptions(key, parentOptions);
+      },
+      commandName
+    );
   }
 
   private processObject(
     objectToProcess: object,
     parentOptions: object,
-    onProcess: (key: string, parentOptions: object) => void
+    onProcess: (key: string, parentOptions: object) => void,
+    commandName: string,
+    path?: string
   ) {
     forEach(objectToProcess, (value, key) => {
+      path = this.resolveObjectPath(key, path);
```

Comment:
We should avoid reassigning parameters and enable the [related eslint rule](https://eslint.org/docs/rules/no-param-reassign).

Reply:
Seems like the entire class faults in this matter. Lets keep your current implementation and address this in a different PR.